pkg/producer/kafka.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2020 VMware, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package producer
+
+import (
+	"github.com/Shopify/sarama"
+	"github.com/golang/protobuf/proto"
+	"k8s.io/klog"
+	"net"
+
+	"github.com/vmware/go-ipfix/pkg/entities"
+	"github.com/vmware/go-ipfix/pkg/producer/protobuf"
+)
+
+var (
+	KafkaConfigVersion sarama.KafkaVersion
+)
+
+type KafkaProducer struct {
+	producer sarama.AsyncProducer
+	topic    string
+}
+
+// convertIPFIXMsgToFlowMsgs converts data records in IPFIX message to flow messages
+// in given proto schema.
+func convertIPFIXMsgToFlowMsgs(msg *entities.Message) []*protobuf.FlowMessage {
+	if msg.Set.GetSetType() == entities.Template {
+		return nil
+	}
```

Comment:
we will not need to send flow message to Kafka consumer for it to decode right?

Reply:
Yes template functionality is already achieved through the defined protobuf schema.